sdk/tests/conformance/canvas/to-data-url-test.html

```diff
@@ -21,29 +21,55 @@
 var wtu = WebGLTestUtils;
 var numTests = 10;
 var gl;
+var glAttributes;
 var ctx;
 
 var main = function() {
   description();
   ctx = document.getElementById("c2d").getContext("2d");
-  gl = wtu.create3DContext("c3d");
 
-  if (!gl) {
-    testFailed("can't create 3d context");
-    finishTest();
-    return;
-  }
+  var updateCanvas = function(width, height, attributes) {
+    var canvas = document.getElementById("c3d");
+
+    if (gl && (glAttributes !== attributes)) {
+      // Attributes changed, recreate the canvas
+      var newCanvas = canvas.cloneNode();
+      canvas.parentNode.replaceChild(newCanvas, canvas);
+      canvas = newCanvas;
+      glAttributes = attributes;
+      gl = undefined;
+    }
+
+    canvas.width = width;
+    canvas.height = height;
+
+    if (!gl) {
+      gl = wtu.create3DContext(canvas, attributes);
+    }
+
+    if (!gl) {
+      testFailed("can't create 3d context");
+      return;
+    }
+
+    return gl;
+  };
 
   var clearRect = function(gl, x, y, width, height, color) {
     gl.clearColor(color[0] / 255, color[1] / 255, color[2] / 255, color[3] / 255);
     gl.scissor(x, y, width, height);
     gl.clear(gl.COLOR_BUFFER_BIT);
   };
 
-  var testSize = function(gl, width, height, callback) {
-    debug("testing " + width + " by " + height);
-    gl.canvas.width = width;
-    gl.canvas.height = height;
+  var testSize = function(width, height, attributes, callback) {
+    var attributesDebugMessage = (attributes ? " attributes: " + JSON.stringify(attributes) : "");
```

Comment:
Parens around compound components of a trinary:
`(attributes ? (" attributes: " + JSON.stringify(attributes)) : "")`
Or more ideally avoid trinaries.